raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -43,16 +43,149 @@ contract Deposit {
 }
 
 
-contract ServiceRegistry is Utils {
-    Token public token;
+contract ServiceRegistryConfigurableParameters {
+    address public controller;
+
+    modifier onlyController() {
+        require(msg.sender == controller);
+        _;
+    }
 
     // After a price is set to set_price at timestamp set_price_at,
     // the price decays according to decayed_price().
     uint256 public set_price;
     uint256 public set_price_at;
 
+    /// The amount of time (in seconds) till the price decreases to roughly 1/e.
+    uint256 public decay_constant = 200 days;
+
     // Once the price is at min_price, it can't decay further.
-    uint256 constant min_price = 1000;
+    uint256 public min_price;
+
+    // Whenever a deposit comes in, the price is multiplied by numerator / denominator.
+    uint256 public price_bump_numerator = 1;
+    uint256 public price_bump_denominator = 1;
+
+    // The duration of service registration/extension in seconds
+    uint256 public registration_duration = 180 days;
+
+    function change_parameters(
+            uint256 _price_bump_numerator,
+            uint256 _price_bump_denominator,
+            uint256 _decay_constant,
+            uint256 _min_price,
+            uint256 _registration_duration
+    ) public onlyController {
```

Comment:
This is an external function call and thus should return a `bool`.

Reply:
Yes, it's hard for me to follow what I say.